### ISSUE TITLE
CLI commands: database drop throws ObjectDisposedException

### ISSUE THREAD

**Issue by flensrocker** [aspnet/EntityFrameworkCore#10693]
When running `dotnet ef database drop -f` an `ObjectDisposedException` is thrown, when the `DbContext` is added to dependency injection with a scoped lifetime.

```
Exception message:
System.ObjectDisposedException: Cannot access a disposed object. A common cause of this error is disposing a context that was resolved from dependency injection and then later trying to use the same context instance elsewhere in your application. This may occur if you are calling Dispose() on the context, or wrapping the context in a using statement. If you are using dependency injection, you should let the dependency injection container take care of disposing context instances.
Objektname: "ApplicationDbContext".
   bei Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   bei Microsoft.EntityFrameworkCore.DbContext.Microsoft.EntityFrameworkCore.Infrastructure.IInfrastructure<System.IServiceProvider>.get_Instance()
   bei Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   bei Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   bei Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   bei Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.DropDatabase(String contextType)
   bei Microsoft.EntityFrameworkCore.Design.OperationExecutor.DropDatabaseImpl(String contextType)
   bei Microsoft.EntityFrameworkCore.Design.OperationExecutor.DropDatabase.<>c__DisplayClass0_1.<.ctor>b__0()
   bei Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Cannot access a disposed object. A common cause of this error is disposing a context that was resolved from dependency injection and then later trying to use the same context instance elsewhere in your application. This may occur if you are calling Dispose() on the context, or wrapping the context in a using statement. If you are using dependency injection, you should let the dependency injection container take care of disposing context instances.
Objektname: "ApplicationDbContext".
```

When printing the stack trace in the constructor and dispose methods, it seems that the first DbContext is created at `DbContextOperations.GetContextInfo`, where it is also disposed (it's wrapped in a `using`).

In `DbContextOperatios.DropDatabase` another DbContext is created and it seems that the already disposed DbContext from `GetContextInfo` is returned.

### Steps to reproduce
Repository: https://github.com/flensrocker/EFCoreIssue

Run `dotnet ef database drop -f` in the subdirectory `src/EFCoreIssue`.

### Further technical details
EF Core version: 1.1.5
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: Visual Studio 2017 15.5.3

**Comment by bricelam** [comment]
Looks like the same root cause as #10636

**Comment by ajcvickers** [comment]
@flensrocker Currently we do not plan to fix this in the 1.1.x codebase. If its is not possible to update to a 2.0.x release, then you might consider a SQL script or a little utility app that uses `context.Database.EnsureDeleted()` as a workaround.

**Comment by flensrocker** [comment]
That's ok with me. There are enough possibilities to drop a database. I plan to upgrade to 2.x, but currently there are problems with angular-end2end tests and protractor. But that's not related to EFCore.

Thanks!
